lib/manager/npm/post-update/yarn.ts

```diff
@@ -21,103 +20,68 @@ export async function generateLockFile(
-        const homeNpmrc = join(homeDir, '.npmrc');
-        cmd += `-v ${homeNpmrc}:/home/ubuntu/.npmrc `;
-      }
-      const envVars = ['NPM_CONFIG_CACHE', 'npm_config_store'];
-      cmd += envVars.map((e) => `-e ${e} `).join('');
-      cmd += `-w "${cwd}" `;
-      cmd += `renovate/yarn yarn`;
+    const preCommands = ['npm i -g yarn'];
+    const commands = [];
+    let cmdOptions = '';
+    if (global.trustLevel !== 'high' || config.ignoreScripts) {
+      cmdOptions = '--ignore-scripts --ignore-engines --ignore-platform';
     }
-
-    let cmdExtras = '';
-    const cmdEnv = { ...env };
-    cmdExtras += ' --ignore-scripts';
-    cmdExtras += ' --ignore-engines';
-    cmdExtras += ' --ignore-platform';
-    const installCmd = cmd + ' install' + cmdExtras;
-    // TODO: Switch to native util.promisify once using only node 8
-    await exec(installCmd, {
+    const execOptions: ExecOptions = {
       cwd,
-      env: cmdEnv,
-    });
+      extraEnv: {
+        NPM_CONFIG_CACHE: process.env.NPM_CONFIG_CACHE,
+        npm_config_store: process.env.npm_config_store,
+      },
+      docker: {
+        image: 'renovate/node',
+        preCommands,
+      },
+    };
+    if (config.dockerMapDotfiles) {
+      const homeDir =
+        process.env.HOME || process.env.HOMEPATH || process.env.USERPROFILE;
+      const homeNpmrc = join(homeDir, '.npmrc');
+      execOptions.docker.volumes = [[homeNpmrc, '/home/ubuntu/.npmrc']];
+    }
+
+    // This command updates the lock file based on package.json
+    commands.push(`yarn install ${cmdOptions}`.trim());
+
+    // rangeStrategy = update-lockfile
     const lockUpdates = upgrades
       .filter((upgrade) => upgrade.isLockfileUpdate)
       .map((upgrade) => upgrade.depName);
     if (lockUpdates.length) {
       logger.debug('Performing lockfileUpdate (yarn)');
-      const updateCmd =
-        cmd +
-        ' upgrade' +
-        lockUpdates.map((depName) => ` ${depName}`).join('') +
-        cmdExtras;
-      const updateRes = await exec(updateCmd, {
-        cwd,
-        env,
-      });
-      // istanbul ignore next
-      stdout += updateRes.stdout || '';
-      stderr += updateRes.stderr || '';
+      commands.push(
+        `yarn upgrade ${lockUpdates.join(' ')} ${cmdOptions}`.trim()
+      );
     }
 
-    if (
-      config.postUpdateOptions &&
-      config.postUpdateOptions.includes('yarnDedupeFewer')
-    ) {
+    // postUpdateOptions
+    if (config.postUpdateOptions?.includes('yarnDedupeFewer')) {
       logger.debug('Performing yarn dedupe fewer');
-      const dedupeCommand =
-        'npx yarn-deduplicate@1.1.1 --strategy fewer && yarn';
-      const dedupeRes = await exec(dedupeCommand, {
-        cwd,
-        env,
-      });
-      // istanbul ignore next
-      stdout += dedupeRes.stdout || '';
-      stderr += dedupeRes.stderr || '';
+      commands.push('npx yarn-deduplicate@2.0.0 --strategy fewer');
```

Comment:
I thinnk we should make the version configurable, or at least as const in module scope

Reply:
What's better for now - a fixed version or no version at all?

Reply:
I think better no version at all and make it configurable later